Add crypto_pwhash to soddium.i

bufke opened the pull request.

joshjdevl commented on src/main/java/org/libsodium/jni/Sodium.java:
```diff
@@ -737,10 +737,6 @@ public static int crypto_onetimeauth_poly1305_verify(byte[] h, byte[] in, int in
     return SodiumJNI.crypto_onetimeauth_poly1305_verify(h, in, inlen, k);
   }
 
-  public static int crypto_onetimeauth_poly1305_statebytes() {
```
Wondering was this removed in a recent version of libsodium?

bufke replied:
I'm not sure. It was auto generated from sodium.i. 

https://github.com/jedisct1/libsodium/commit/f1fdb2a763b89c6321a24307603fa10d552794c7

Is it possible I am using an older version of libsodium somehow? I can just remove this change if you'd like. It has nothing to do with my intended changes.

joshjdevl replied:
Could be a different version. Maybe try running 'submodule-update.sh' to update the libsodium version? That will bring to the latest version of branch 'stable'

joshjdevl commented:
Thanks! This is helpful and looks good! Just one comment which I added.

bufke commented:
Ok I reverted the poly1305_statebytes deletions. I still don't know why they happen though.

I also squashed my commits (miss gitlab's gui for this!) 

Let me know if you need anything else.
